compiler/src/dotty/tools/dotc/interactive/Completion.scala

```diff
@@ -112,10 +112,11 @@ object Completion {
     val completer = new Completer(mode, prefix, pos)
 
     val completions = path match {
-        case Select(qual, _) :: _                              => completer.selectionCompletions(qual)
-        case Import(expr, _) :: _                              => completer.directMemberCompletions(expr)
-        case (_: untpd.ImportSelector) :: Import(expr, _) :: _ => completer.directMemberCompletions(expr)
-        case _                                                 => completer.scopeCompletions
+        case Select(qual @ This(_), _) :: _ if qual.span.isSynthetic => completer.scopeCompletions
```

Comment:
Since it's non-obvious why this would be needed, I suggest adding a comment with an example here (or a reference to the added test case).

Reply:
Added